### ISSUE TITLE
User migration (export/import data via user migration app)

### ISSUE THREAD

**Issue by juliusknorr** [nextcloud/deck#3828]
With 24 we have a server API to handle user migration which deck should also support by implementing https://docs.nextcloud.com/server/stable/developer_manual/digging_deeper/user_migration.html

Related App: https://apps.nextcloud.com/apps/user_migration

We already have a way to dump json exports of boards which could be reused, but it might be missing some data:
- Comments
- Activity
- File shares

Especially with file shares it might be a topic for further discussion on how to handle those on user migration. The files might not be owned by the board owner and therefore would not be migrated. Also sharing migration is currently not implemented for files in general.

**Comment by Olen** [comment]
What is needed to get this running? Is there anything I can do?
We already have both export and import of deck data (unfortunately the import script can't import the data that the export script generates...) so it should be possible to reuse a lot of that code.
But I don't want to duplicate any work if someone is already working on this.

**Comment by ne0YT** [comment]
@Olen were you able to figure this out?

**Comment by Olen** [comment]
I don't remember exactly what I did. I think I made an export, and then modified it manually to create a valid import file, and then imported it.  
Or did I simply take a database dump, adjust some values, and push it to the new db?  
It's not somethng I do on a regular basis, so for the one time, I just made something that worked there and then

**Comment by politas** [comment]
I just gave up and recreated my decks manually.

**Comment by paskalito** [comment]
@ne0YT 

see here: https://github.com/nextcloud/deck/issues/3506#issuecomment-2416265703

interested in hearing back if it worked for you

**Comment by ne0YT** [comment]
@paskalito it doesn't work for me on the snap package.
it always tells me "config file not found", no matter if I use --config or if i pass it when it asks for it using the full path or ./thefile.json etc.
this might be a snap specific thing, idk.

**Comment by ne0YT** [comment]
yeah the file obviously needs to be accessible from within snap then it works :) also tested with AIO

**Comment by RainerKlute** [comment]
Migrating data from one Nextcloud instance to another instance (not to mention portable exports) as well as from one user to another is **always** an issue, at least potentially. I may be wrong, but my general impression is that this is given surprisingly little consideration in Nextcloud apps.

**Comment by Markoise** [comment]
> Migrating data from one Nextcloud instance to another instance (not to mention portable exports) as well as from one user to another is **always** an issue, at least potentially. I may be wrong, but my general impression is that this is given surprisingly little consideration in Nextcloud apps.

Yes, that's a very big problem for anyone who doesn't have or can't run their own server.
So most apps for such users are not for serious use. Unless the stored data is not important.

**Comment by jospoortvliet** [comment]
> > Migrating data from one Nextcloud instance to another instance (not to mention portable exports) as well as from one user to another is **always** an issue, at least potentially. I may be wrong, but my general impression is that this is given surprisingly little consideration in Nextcloud apps.
> 
> Yes, that's a very big problem for anyone who doesn't have or can't run their own server. So most apps for such users are not for serious use. Unless the stored data is not important.

Moving an entire server is quite doable - export data & database, import on other system. But per-user is indeed in need of some love!
